src/celttf/truetypefont.cpp

```diff
@@ -646,69 +708,89 @@ TextureFont* TextureFont::load(const Renderer *r, const fs::path &path, int inde
     if (!FT_IS_SCALABLE(face))
     {
         GetLogger()->error("Font is not scalable: {}\n", path);
+        FT_Done_Face(face);
         return nullptr;
     }
 
     if (FT_Set_Char_Size(face, 0, size << 6, dpi, dpi) != 0)
     {
         GetLogger()->error("Could not set font size {}\n", size);
+        FT_Done_Face(face);
         return nullptr;
     }
 
-    auto* font = new TextureFont(r);
-    font->impl->m_face = face;
-
-    if (!font->impl->buildAtlas())
-        return nullptr;
-
-    font->setMaxAscent(face->size->metrics.ascender >> 6);
-    font->setMaxDescent(-face->size->metrics.descender >> 6);
-
-    return font;
+    return face;
 }
 
 // temporary while no fontconfig support
-static fs::path ParseFontName(const fs::path &filename, int &collectionIndex, int &size)
+fs::path
+ParseFontName(const fs::path &filename, int &index, int &size)
 {
     // Format with font path/collection index(if any)/font size(if any)
     auto fn = filename.string();
-    auto pos = fn.rfind(',');
-    if (pos != string::npos)
+    if (auto ps = fn.rfind(','); ps != std::string::npos)
```

Comment:
Better to combine the conditions with `&&` to avoid the indentation

Reply:
what do you mean?

```
    if (auto ps = fn.rfind(','); ps != std::string::npos &&
        from_chars(&fn[ps + 1], &fn[fn.size()], size).ec == std::errc())
    {
```
this?

Reply:
Never mind, forgot that this doesn't work with the bindings.